src/detect-lua.h

```diff
@@ -49,6 +49,8 @@ typedef struct DetectLuaData {
     uint32_t sid;
     uint32_t rev;
     uint32_t gid;
+
+    DetectEngineCtx *de_ctx;
```

Comment:
Would be nice if we can avoid this, at least in the git master. See also https://github.com/OISF/suricata/pull/4716#discussion_r396017441

Reply:
The issue's occurring because the signature is deemed invalid after the lua keyword has been processed. I need to locate the previously registered item and mark it as invalid. The only way to do that (correct me if I'm wrong), is to locate it from the `de_ctx->keyword_list`. 

I could encode the `keyword_list` into a value returned by `DetectRegisterThreadCtxFuncs` or keep track of it in a container local to `detect-lua.c` ... I don't like either of these options. Are there others?

Reply:
I think it would make sense to give this `Free` callback another arg: `de_ctx`. Then you can do the same thing as now w/o keeping this pointer here.

Reply:
The `Free` function of `SigTableElmt`?

Reply:
Yeah